File: frontend/src/types/VolumeInfo.ts
import {IndustryIdentifier} from "./IndustryIdentifier.ts";
import {ImageLink} from "./ImageLink.ts";

export type VolumeInfo = {
    title: string,
    authors: string[],
    publisher: string,
    industryIdentifiers: IndustryIdentifier[],
    categories: string[],
    imageLinks: ImageLink,
    description: string
}

File: backend/src/main/java/org/schmaelzle/backend/google/VolumeInfo.java
package org.schmaelzle.backend.google;

import java.util.List;

public record VolumeInfo(
        String title,
        List<String> authors,
        String publisher,
        List<IndustryIdentifier> industryIdentifiers,
        List<String> categories,
        ImageLink imageLinks,
        String description
) {
}
